File: frontend/web/features/management/types/teacher.ts
export type TeacherValues = Record<string, string> & {
  photoDataUrl: string;
  teacherCode: string;
  name: string;
  email: string;
  departmentId?: string;
  department?: string;
  status: string;
  createAt: string;
};

export type TeacherItem = { id: string; values: TeacherValues };

File: backend/src/InstituteManagement.Domain/Entities/Teacher.cs
namespace InstituteManagement.Domain.Entities;

public sealed class Teacher : Entity
{
    public required string TeacherNumber { get; set; }
    public required string FullName { get; set; }
    public string Email { get; set; } = string.Empty;
    public string PhotoDataUrl { get; set; } = string.Empty;
    public Guid DepartmentId { get; set; }
    public Department? Department { get; set; }
    public string Status { get; set; } = "Available";
}
